packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -302,20 +302,105 @@ List<Plugin> findPlugins(FlutterProject project) {
   }
   packages.forEach((String name, Uri uri) {
     final Uri packageRoot = uri.resolve('..');
-    final Plugin plugin = _pluginFromPubspec(name, packageRoot);
+    final Plugin plugin = _pluginFromPackage(name, packageRoot);
     if (plugin != null) {
       plugins.add(plugin);
     }
   });
   return plugins;
 }
 
+void _addPluginsToPlatform(List<Plugin> plugins, FlutterProjectPlatform platform, Map<String, dynamic> pluginsMap) {
+  if (!platform.existsSync()) {
+    return;
+  }
+  final List<Map<String,dynamic>> list = platform.pluginsList(plugins);
+  if (list.isNotEmpty) {
+    pluginsMap[platform.pluginConfigKey] = list;
+  }
+}
+
+/// Writes the .flutter-plugins.json file based on the list of plugins.
+/// If there aren't any plugins, then the files aren't written to disk. The resulting
+/// file looks something like this:
+/// {
+///     "_info": "// This is a generated file; do not edit or check into version control.",
```

Comment:
I would also recommend embedding some sort of version string in the JSON, just in case

Reply:
Also possibly the date it was generated on/ the version of the tool used to generate it

Reply:
See .dart_tool/package_config.json:

```
  "generated": "2020-01-13T22:42:48.743606Z",
  "generator": "pub",
  "generatorVersion": "2.8.0-dev.0.0.flutter-395daaa3ec"
```

Reply:
Done